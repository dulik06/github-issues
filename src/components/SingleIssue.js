import React, { Component } from 'react';
import PropTypes from 'prop-types';
import Labels from './Labels';
import Timestamp from 'react-timestamp';

import openIssueSign from '../assets/svg/attention-sign-outline.svg'

class SingleIssue extends Component {

  static propTypes = {
    title: PropTypes.string.isRequired,
    number: PropTypes.number.isRequired,
    labels: PropTypes.array.isRequired,
    time: PropTypes.string.isRequired,
  }

  render() {
    return (
      <li>
        <div className='single-issue__title--container'>
          <img src={openIssueSign} alt='open issues sign' className='single-issue__title--icon'/>
          <div className='single-issue__title--text'>{this.props.title}</div>
        </div>
        <div className='single-issue__info'>
          <span className='single-issue__info--text'>
            #{this.props.number} opened
            <Timestamp time={this.props.time}/>
            by {this.props.author}
          </span>
         {
           this.props.labels.length > 0 ? 
            this.props.labels.map(label =>
              <Labels name={label.name} color={label.color} key={label.id} id="labels"/>
            ) 
            :
            null
          }
        </div>
      </li>
     
    )
  }
}
export default SingleIssue;