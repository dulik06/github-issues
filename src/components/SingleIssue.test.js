import React from 'react';
import { shallow } from 'enzyme';

import SingleIssue from './SingleIssue';
import openIssueSign from '../assets/svg/attention-sign-outline.svg'

const props = {
  title: "test title",
  number: 2018,
  labels: [],
  time: "2018-11-18T22:53:22Z"
}
describe('SingleIssue', () => {
  const wrapper = shallow(<SingleIssue {...props} />);

  it('should render without crashing', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('should not display <Labels /> component if labels array is empty', () => {
    expect(wrapper.find('#labels')).toHaveLength(0);
  });

  it('should display the open issues sign', () => {
    expect(wrapper.find('img').prop('src')).toEqual(openIssueSign);
  });

  it('should display the issue title', () => {
    expect(wrapper.find('div.single-issue__title--text').text()).toEqual("test title");
  });

  it('should display issue details', () => {
    expect(wrapper.find('div.single-issue__info')).toHaveLength(1);
  });
})