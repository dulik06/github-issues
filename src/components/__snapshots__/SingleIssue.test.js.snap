// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SingleIssue should render without crashing 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <SingleIssue
    labels={Array []}
    number={2018}
    time="2018-11-18T22:53:22Z"
    title="test title"
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <div
          className="single-issue__title--container"
        >
          <img
            alt="open issues sign"
            className="single-issue__title--icon"
            src="attention-sign-outline.svg"
          />
          <div
            className="single-issue__title--text"
          >
            test title
          </div>
        </div>,
        <div
          className="single-issue__info"
        >
          <span
            className="single-issue__info--text"
          >
            #
            2018
             opened
            <Timestamp
              actualSeconds={false}
              autoUpdate={false}
              className=""
              component="span"
              format="ago"
              includeDay={false}
              precision={1}
              since={null}
              style={Object {}}
              time="2018-11-18T22:53:22Z"
              twentyFourHour={false}
              until={null}
              utc={true}
            />
            by 
          </span>
        </div>,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <img
              alt="open issues sign"
              className="single-issue__title--icon"
              src="attention-sign-outline.svg"
            />,
            <div
              className="single-issue__title--text"
            >
              test title
            </div>,
          ],
          "className": "single-issue__title--container",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "alt": "open issues sign",
              "className": "single-issue__title--icon",
              "src": "attention-sign-outline.svg",
            },
            "ref": null,
            "rendered": null,
            "type": "img",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "test title",
              "className": "single-issue__title--text",
            },
            "ref": null,
            "rendered": "test title",
            "type": "div",
          },
        ],
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <span
              className="single-issue__info--text"
            >
              #
              2018
               opened
              <Timestamp
                actualSeconds={false}
                autoUpdate={false}
                className=""
                component="span"
                format="ago"
                includeDay={false}
                precision={1}
                since={null}
                style={Object {}}
                time="2018-11-18T22:53:22Z"
                twentyFourHour={false}
                until={null}
                utc={true}
              />
              by 
            </span>,
            null,
          ],
          "className": "single-issue__info",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                "#",
                2018,
                " opened",
                <Timestamp
                  actualSeconds={false}
                  autoUpdate={false}
                  className=""
                  component="span"
                  format="ago"
                  includeDay={false}
                  precision={1}
                  since={null}
                  style={Object {}}
                  time="2018-11-18T22:53:22Z"
                  twentyFourHour={false}
                  until={null}
                  utc={true}
                />,
                "by ",
                undefined,
              ],
              "className": "single-issue__info--text",
            },
            "ref": null,
            "rendered": Array [
              "#",
              2018,
              " opened",
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "actualSeconds": false,
                  "autoUpdate": false,
                  "className": "",
                  "component": "span",
                  "format": "ago",
                  "includeDay": false,
                  "precision": 1,
                  "since": null,
                  "style": Object {},
                  "time": "2018-11-18T22:53:22Z",
                  "twentyFourHour": false,
                  "until": null,
                  "utc": true,
                },
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
              "by ",
              undefined,
            ],
            "type": "span",
          },
          null,
        ],
        "type": "div",
      },
    ],
    "type": "li",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <div
            className="single-issue__title--container"
          >
            <img
              alt="open issues sign"
              className="single-issue__title--icon"
              src="attention-sign-outline.svg"
            />
            <div
              className="single-issue__title--text"
            >
              test title
            </div>
          </div>,
          <div
            className="single-issue__info"
          >
            <span
              className="single-issue__info--text"
            >
              #
              2018
               opened
              <Timestamp
                actualSeconds={false}
                autoUpdate={false}
                className=""
                component="span"
                format="ago"
                includeDay={false}
                precision={1}
                since={null}
                style={Object {}}
                time="2018-11-18T22:53:22Z"
                twentyFourHour={false}
                until={null}
                utc={true}
              />
              by 
            </span>
          </div>,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <img
                alt="open issues sign"
                className="single-issue__title--icon"
                src="attention-sign-outline.svg"
              />,
              <div
                className="single-issue__title--text"
              >
                test title
              </div>,
            ],
            "className": "single-issue__title--container",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "alt": "open issues sign",
                "className": "single-issue__title--icon",
                "src": "attention-sign-outline.svg",
              },
              "ref": null,
              "rendered": null,
              "type": "img",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "test title",
                "className": "single-issue__title--text",
              },
              "ref": null,
              "rendered": "test title",
              "type": "div",
            },
          ],
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <span
                className="single-issue__info--text"
              >
                #
                2018
                 opened
                <Timestamp
                  actualSeconds={false}
                  autoUpdate={false}
                  className=""
                  component="span"
                  format="ago"
                  includeDay={false}
                  precision={1}
                  since={null}
                  style={Object {}}
                  time="2018-11-18T22:53:22Z"
                  twentyFourHour={false}
                  until={null}
                  utc={true}
                />
                by 
              </span>,
              null,
            ],
            "className": "single-issue__info",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  "#",
                  2018,
                  " opened",
                  <Timestamp
                    actualSeconds={false}
                    autoUpdate={false}
                    className=""
                    component="span"
                    format="ago"
                    includeDay={false}
                    precision={1}
                    since={null}
                    style={Object {}}
                    time="2018-11-18T22:53:22Z"
                    twentyFourHour={false}
                    until={null}
                    utc={true}
                  />,
                  "by ",
                  undefined,
                ],
                "className": "single-issue__info--text",
              },
              "ref": null,
              "rendered": Array [
                "#",
                2018,
                " opened",
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "class",
                  "props": Object {
                    "actualSeconds": false,
                    "autoUpdate": false,
                    "className": "",
                    "component": "span",
                    "format": "ago",
                    "includeDay": false,
                    "precision": 1,
                    "since": null,
                    "style": Object {},
                    "time": "2018-11-18T22:53:22Z",
                    "twentyFourHour": false,
                    "until": null,
                    "utc": true,
                  },
                  "ref": null,
                  "rendered": null,
                  "type": [Function],
                },
                "by ",
                undefined,
              ],
              "type": "span",
            },
            null,
          ],
          "type": "div",
        },
      ],
      "type": "li",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
